### PR TITLE
:book: Add note about v1alpha3 removal to book

### PR DESCRIPTION
**Pull request by killianmuldoon**
Add a message to the note in the book about API version removal detailing the issues that may be caused in garbage collection and the actions necessary for fixing the issue.

/area documentation

Part of https://github.com/kubernetes-sigs/cluster-api/issues/8038

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -179,14 +179,20 @@ Cluster API maintains the most recent release/releases for all supported API and
 
 We are going to remove the apiVersions in upcoming releases:
 * v1.5:
-  * Kubernetes API server will stop serving the v1alpha3 apiVersion
+  * Kubernetes API server has stopped serving the v1alpha3 apiVersion
```
Comment (sbueringer) [inline CONTRIBUTING.md:182]: just a nit, wondering if we can phrase this and the next few in a way that it's neutral to if we already did it or not

Feel free to just resolve

**Review by sbueringer — COMMENTED** [review]
/lgtm

File: CONTRIBUTING.md (hunk 1 above)
Comment (killianmuldoon) [reply CONTRIBUTING.md:182]: Good idea - I've updated the wording now.

**Comment by sbueringer** [comment]
Thx!

/lgtm
/approve

**Pull request merged**